### DISCUSSION TITLE
Prevent closing a popover when opening a modal by clicking the modal's trigger in the popover

### DISCUSSION THREAD

**Discussion by mattp0123** [chakra-ui/chakra-ui#6481] [Q&A] [answered]
https://codesandbox.io/s/chakra-modal-in-popover-w3fs8k?file=/src/App.tsx
I want the popover to keep open when I open the modal inside the popover.
What should I do?

**Comment by FlorianMrn** [comment, marked as answer]
Solution found: add closeOnBlur={false} prop on Popover.

**Reply by imMatheus** [reply]
An issue with this approach is that you loose what `closeOnBlur` does, which is that it closes the popover if you click outside of it (which is almost always what you want I feel). Say for example you have a popover shows some buttons with different actions, and one of them is a "delete" button that should open a modal, then the you might still want the `closeOnBlur` effect to exist. An alternative is to have both popover and modal be controlled with two different `useDisclosure` and then check that the modal is not open before closing the popover. Here is an example that solves the same problem but keeps the `closeOnBlur` effect:

```tsx

function MyComponent() {
  const popover = useDisclosure()
  const modal = useDisclosure()

  return (
    <Popover
      isOpen={popover.isOpen}
      onClose={() => {
        if (!modal.isOpen) { // <- The important check
          popover.onClose()
        }
      }}
    >
      <PopoverTrigger>
        <Button onClick={popover.onOpen}>Open Popover</Button>
      </PopoverTrigger>
      <PopoverContent>
        <Button {...modal.getButtonProps()}>Open Modal</Button>
        <Modal isOpen={modal.isOpen} onClose={modal.onClose}>
          <ModalOverlay />
          <ModalContent>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Ducimus,
            repellat.
          </ModalContent>
        </Modal>
      </PopoverContent>
    </Popover>
  )
}
```
